website/docs/r/configuration_entry.markdown

```diff
@@ -0,0 +1,71 @@
+---
+layout: "consul"
+page_title: "Consul: configuration_entry"
+sidebar_current: "docs-consul-resource-configuration-entry"
+description: |-
+  Registers a configuration entry in Consul.
+---
+
+# consul_configuration_entry
+
+The [Configuration Entry](https://www.consul.io/docs/agent/config_entries.html)
+resource can be used to provide cluster-wide defaults for various aspects of
+Consul.
+
+## Example Usage
+
+```hcl
+resource "consul_configuration_entry" "service-defaults" {
+	name = "foo"
+	kind = "service-defaults"
+
+	protocol = "https"
+}
+```
+
+```hcl
+resource "consul_configuration_entry" "proxy-defaults" {
+	name = "global"
+	kind = "proxy-defaults"
+
+	config = {
+        local_connect_timeout_ms = 1000
+        handshake_timeout_ms = 1000
+	}
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `kind` - (Required) The kind of configuration entry to register. Can be
+`proxy-defaults` or `service-defaults`.
+
+* `name` - (Required) The name of the configuration entry being registred. If
+`kind` is `proxy-defaults`, `name` must be `global`.
+
+* `config` - (Optional) An arbitrary map of configuration values used by Connect
+proxies. Can only be set when `kind` is `proxy-defaults`.
+
+* `protocol` - (Optional) The protocol of the service. Can only be set when
+`kind` is `service-defaults`.
```

Comment:
I suggest if we can make this generic that we just document the kind and name and then link to the https://www.consul.io/docs/agent/config_entries.html#configuration-entry-kinds docs for details so we don't have to remember to keep this in sync too across new kinds.